planner/min_priority_queue.go

```diff
@@ -0,0 +1,27 @@
+package planner
+
+type MinPriorityQueue []Vertex
+
+func (pq MinPriorityQueue) Len() int {
+	return len(pq)
+}
+
+func (pq MinPriorityQueue) Less(i, j int) bool {
+	return pq[i].Key <= pq[j].Key
+}
+
+func (pq MinPriorityQueue) Swap(i, j int) {
+	pq[i], pq[j] = pq[j], pq[i]
+}
+
+func (pq *MinPriorityQueue) Push(item interface{}) {
```

Comment:
Does the new data structure `MinPriorityQueue` keep the min-heap data structure? for example, the first element is always the minimum?

Reply:
It's not a new data structure. Just renamed it.

Reply:
I see, thanks.